filesystem/storage/CacheGeneratedAssetHandler.php

```diff
@@ -0,0 +1,167 @@
+<?php
+
+namespace SilverStripe\Filesystem\Storage;
+
+use Config;
+use Exception;
+use Flushable;
+use SS_Cache;
+use Zend_Cache_Core;
+
+/**
+ * Handle references to generated files via cached tuples
+ *
+ * Important: If you are using the default FlysystemStore with legacy_filenames, you will need to ?flush
+ * in order to refresh combined files.
+ * 
+ * @package framework
+ * @subpackage filesystem
+ */
+class CacheGeneratedAssetHandler implements GeneratedAssetHandler, Flushable {
+
+	/**
+	 * Lifetime of cache
+	 *
+	 * @config
+	 * @var int
+	 */
+	private static $lifetime = 0;
+
+	/**
+	 * Backend for generated files
+	 *
+	 * @var AssetStore
+	 */
+	protected $assetStore = null;
+
+	/**
+	 * Assign the asset backend
+	 *
+	 * @param AssetStore $store
+	 * @return $this
+	 */
+	public function setAssetStore(AssetStore $store) {
+		$this->assetStore = $store;
+		return $this;
+	}
+
+	/**
+	 * Get the asset backend
+	 *
+	 * @return AssetStore
+	 */
+	public function getAssetStore() {
+		return $this->assetStore;
+	}
+
+	/**
+	 * @return Zend_Cache_Core
+	 */
+	protected static function get_cache() {
+		$cache = SS_Cache::factory('CacheGeneratedAssetHandler');
+		$cache->setLifetime(Config::inst()->get(__CLASS__, 'lifetime'));
+		return $cache;
+	}
+
+	/**
+	 * Flush the cache
+	 */
+	public static function flush() {
+		self::get_cache()->clean();
```

Comment:
I'd expect this to delete the underlying files as well, not only the cached references to them